integr8ly/ci-cd
# Intly 2235 add credentials

## What
The following changes are done as part of the work to create a Jenkins configuration scripts to include all the required delorean configuration. This PR includes the required changes to configure credentials in an existing Jenkins instance.

- [x] Create a role for adding credentials to an existing jenkins
      - It should be able to create user/pass and ssh user with private key credentials.
- [x] Create a `configure_jenkins` playbook to configure an existing jenkins with delorean required resources.

JIRA: https://issues.jboss.org/browse/INTLY-2235

## Verification
1. Run the script with the following inventory file in the `inventories/group_vars/all` folder : 
```
---
jenkins_url: "<jenkins-url>"
jenkins_admin_username: "<userid>"
jenkins_admin_password: "<password/token>"
jenkins_csrf_enabled: "<true/false>" # If CSRF prevention is enabled

jenkins_user_pass_credentials:
  - test-user-pass:
      username: "test-username"
      password: "test-password"
      description: "Sample user pass credential"

jenkins_ssh_user_key_credentials:
  - test-ssh-with-key:
      username: "test-username"
      private_key: |
        SamplePrivateKey
      passphrase: "test"
      description: "Sample SSH user with private key"
```
2. Verify that all credentials have been created successfully
